examples/defaulter-gen/_output_tests/marker/type.go

```diff
@@ -24,17 +24,35 @@ type Defaulted struct {
 	empty.TypeMeta
 
 	// +default="bar"
-	Field string `json:"Field,omitempty"`
-	// +default=0
-	OtherField int `json:"OtherField,omitempty"`
-
-	// Default is forced to 0
-	EmptyInt int
+	StringDefault string `json:"omitempty"`
 
 	// Default is forced to empty string
 	// Specifying the default is a no-op
 	// +default=""
-	EmptyString string
+	StringEmptyDefault string
+
+	// Not specifying a default still defaults for non-omitempty
+	StringEmpty string
+
+	// +default="default"
```

Comment:
What should these do?

// +default="de\nfault"
// +default="de\"fault"
// +default="de"fault"

Reply:
`// +default="de\nfault"`: Code generation passes because this is a valid assignment to a string
`// +default="de\"fault"`: Passes because this is a valid assignment to a string
`// +default="de"fault"`: Fails because this breaks our syntax

We attempt an unmarshal in the code before generation. If the unmarshal fails, we return an error and do not generate the code. If the unmarshal is successful, we pretty much assignment the string value verbatim. Eg:

```
    if in.StringDefault == "" {
        in.StringDefault = "de\"fault"
    }
```

Reply:
I think another way to ask this question is: are all json valid strings also valid (and corresponding) go strings? What about "`"?

Reply:
I guess I was looking for a test that exercised these. I see a call to strconv.Quote but no Unquote anywhere.

Combined with Antoine's point, I think we should say that the default is using JSON strings, and somewhere I should see a place that unmarshalls via JSON and then calls strconv.Quote to get a go-safe form.

Also I can't seem to find the place that would be returning an error to users? Is it preexisting?

Reply:
Although I'm realizing these questions can be fixed in a follow-up.

Reply:
> I think we should say that the default is using JSON strings, and somewhere I should see a place that unmarshalls via JSON and then calls strconv.Quote to get a go-safe form.

It's definitely supposed to be json strings in input, and yes normalizing the string somehow is the way to go, thanks!

Reply:
Originally there were suggestions around normalizing the string to ensure the input is valid. Right now, the code does a one way unmarshal, and if the unmarshal is successful we use the original string without normalization. I can't really think of a test case where a one way unmarshal would pass but a string normalization would result in a different string. I'd be happy to make the change, just want to make sure it's something we need.